src/make.ts

```diff
@@ -326,14 +328,32 @@ export async function doBuildTarget(progress: vscode.Progress<{}>, target: strin
             logger.messageNoCR(result, "Normal");
         };
 
-        // The build invocation should use the system locale.
-        const result: util.SpawnProcessResult = await util.spawnChildProcess(configuration.getConfigurationMakeCommand(), makeArgs, util.getWorkspaceRoot(), false, stdout, stderr);
-        if (result.returnCode !== ConfigureBuildReturnCodeTypes.success) {
+        let myTaskCommand: string = `${configuration.getConfigurationMakeCommand()} ${makeArgs.join(" ")}`;
```

Comment:
Does `makeArgs.join` do the right thing for args with spaces in them or do we need to surround spaced args with quotes before doing this?  Alternately, is there a function on the `shlex` module that can compose the command line for us?  Or does the `vscode.ShellExecution` support passing in the args as an array so we don't have to do it?

Reply:
Great idea, found that ShellExecution can make a clear separation between command (which is make) and args and has settings to properly quote each of them.